pkg/provisioner/ironic/ironic.go

```diff
@@ -1336,6 +1340,34 @@ func (p *ironicProvisioner) Delete() (result provisioner.Result, err error) {
 		return p.setMaintenanceFlag(ironicNode, true)
 	}
 
+	if ironicNode.PowerState == powerOn {
+		p.log.Info("host is being powered off before removal")
+		provResult, err := p.hardPowerOff()
+		if err != nil {
+			switch err.(type) {
+			case HostLockedError:
+				result.RequeueAfter = powerRequeueDelay
+				result.Dirty = true
+				return result, err
+			default:
+				if result.ErrorCount < maxErrorRetryCount {
+					result.ErrorCount++
+					result.Dirty = true
+					return result, err
+				}
+				result.Dirty = false
```

Comment:
This is the default value. Are you resetting it because it was changed somewhere else?

Reply:
Again, this line to reset the dirty flag was added because I thought we were preserving the result structure during subsequent calls to the same method. Removing it since a new result structure is used for each call.